code/mob/living/critter/small_animal.dm

```diff
@@ -677,6 +688,43 @@ ABSTRACT_TYPE(/mob/living/critter/small_animal)
 					src.delStatus("weakened")
 					src.icon_state = src.dogtype
 
+	Life(datum/controller/process/mobs/parent)
+		if (..())
+			return TRUE
+
+		//Dogs bark sometimes
+		if (src.ai?.enabled && prob(5))
+			src.emote("scream", TRUE)
+
+	critter_attack(var/the_target)
+		if (istype(the_target, /mob))
+			var/mob/target = the_target
+			var/datum/targetable/critter/pounce/pounce = src.abilityHolder.getAbility(/datum/targetable/critter/pounce)
+			if (!pounce.disabled && pounce.cooldowncheck() && prob(50))
+				src.visible_message("<span class='combat'><B>[src]</B> barrels into [target] and trips them!</span>", "<span class='combat'>You run into [target]!</span>")
+				pounce.handleCast(target)
+				return
+			src.visible_message("<span class='combat'><B>[src]</B> bites [target]!</span>", "<span class='combat'>You bite [target]!</span>")
+			playsound(src.loc, 'sound/items/eatfoodshort.ogg', 50, 1, -1, 1.3)
+			random_brute_damage(target, src.attack_damage, 1)
+
+	disposing()
+		. = ..()
+		UnregisterSignal(src, COMSIG_MOB_THROW_ITEM_NEARBY)
+
+	proc/throw_response(target, item, thrower)
```

Comment:
```suggestion
	proc/throw_response(target, obj/item/item, mob/thrower)
```
Arguments should be typed- suggestions might be wrong (is a graviton a valid thrower?)